infra-in-containers/resources/deploy.yml

```diff
@@ -0,0 +1,135 @@
+      default: "jiunnyuan@commandocloudlet.com"
+      private: no
+    - name: "email_password"
+      prompt: "account password"
+      private: yes
+      confirm: yes
+    - name: "relay_host"
+      prompt: "What is your email relay host?"
+      default: "smtp.sendgrid.net"
+      private: no
+    - name: "relay_port"
+      prompt: "What is your email relay port?"
+      default: "587"
+      private: no
+    - name: "relay_username"
+      prompt: "What is your email relay username?"
+      default: "apikey"
+      private: no
+    - name: "relay_key"
+      prompt: "What is your email relay password?"
+      default: "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"
+      private: no
+
+  tasks:
+    - name: Ensure {{base_path}} exists
+      file: path={{base_path}} state=directory
+    - name: "download docker-compose.yml"
+      get_url:
+        url: https://raw.githubusercontent.com/CloudCommandos/missions/CC/infra-in-containers/resources/docker-compose.yml
+        dest: "{{base_path}}/docker-compose.yml"
+        mode: 0777
+        force: yes
+    - name: "download .env"
+      get_url:
+        url: https://raw.githubusercontent.com/CloudCommandos/missions/CC/infra-in-containers/resources/env.txt
+        dest: "{{base_path}}/.env"
+        mode: 0777
+        force: yes
+    - name: "download setup.sh"
+      get_url:
+        url: https://raw.githubusercontent.com/CloudCommandos/missions/CC/infra-in-containers/resources/setup.sh
+        dest: "{{base_path}}/setup.sh"
+        mode: 0777
+        force: yes
+
+    - name: "edit domain name in .env"
+      lineinfile:
+        dest: "{{base_path}}/.env"
+        regexp: '^DOMAINNAME='
+        line: 'DOMAINNAME={{domain}}'
+    - name: install dependencies
+      apt:
+        name:
+            - apt-transport-https
+            - ca-certificates
+            - curl
+            - gnupg2
+            - software-properties-common
+            - python-pip
+            - python-apt
+            - openssl
+        state: present
+    - name: install pexpect
+      pip:
+        name: pexpect
+    - name: adding apt-key for docker
+      apt_key:
+        url: https://download.docker.com/linux/debian/gpg
+        state: present
+    - name: adding docker repo list
+      apt_repository:
+        repo: deb [arch=amd64] https://download.docker.com/linux/debian stretch stable
+        state: present
+    - name: install docker-ce and docker-compose
+      apt:
+        name:
+            - docker-compose
+            - docker-ce
+        state: present
+    - name: create directory
+      file:
+        path: "{{base_path}}/ssl"
+        state: directory
+    - name: generate a self signed OpenSSL certificate
+      expect:
+        command: openssl req -new -newkey rsa:4096 -x509 -sha256 -days 365 -nodes -out {{base_path}}/ssl/ssl.crt -keyout {{base_path}}/ssl/ssl.key
+        responses:
+             "Country Name" : ""
```

Comment:
These fields are not very useful today. Avoid using them by generating the cert in `CN` mode. cfssl and EasyRSA generates certs in this mode by default.